packages/server/validators/users.js

```diff
@@ -0,0 +1,23 @@
+const { param } = require("express-validator");
+
+const UserController = require("../controllers/users");
+
+const userIdValidator = [
+  param("userId")
+    .notEmpty()
+    .withMessage("User ID cannot be empty")
+    .blacklist("<>")
+    .isUUID()
+    .withMessage("Invalid user ID provided")
+    .custom(async (value) => {
+      const user = await UserController.getUser(value);
+      if (!user) {
+        throw new Error("User with ID ${value} not found");
+      }
+      return true;
+    }),
```

Comment:
This custom validator shouldn't return anything since this is not the correct use case for that.

From the docs,
```
Custom validators can be asynchronous, in which case it can return a promise.
The returned promise is awaited on, and it must resolve in order for the field to be valid.
If it rejects, the field is deemed invalid.
```